platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/DefaultConfigurationCache.kt

```diff
@@ -258,7 +258,10 @@ class DefaultConfigurationCache internal constructor(
             isStoreSkipped -> {
                 // build work graph without contributing to a cache entry
                 val finalizedGraph = runAtConfigurationTime {
-                    scheduler(graph)
+                    val result = scheduler(graph)
+                    // force computation of degradation reasons at configuration time as it shouldn't be attempted at execution time
+                    problems.shouldDegradeGracefully()
+                    result
                 }
```

Comment:
I believe this is a smell. 

Degradation reasons should be computed via an explicit method intended as such, instead of lazily. It should only be allowed to be invoked during configuration. Finally, attempting to query degradation reasons when they have not be computed yet should be an error.